tests/visualization_tests/test_edf.py

```diff
@@ -1,106 +1,147 @@
-    assert len(figure.data) == 1
-    assert figure.layout.xaxis.title.text == "Objective Value"
-
-    # Test with two studies.
-    study1 = create_study(direction=direction)
-    study1.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
-    figure = plot_edf([study0, study1])
-    for points in figure.data:
-        _validate_edf_values(points["y"])
-    assert len(figure.data) == 2
-    figure = plot_edf((study0, study1))
-    for points in figure.data:
-        _validate_edf_values(points["y"])
-    assert len(figure.data) == 2
-
-    # Test with a customized target value.
-    study0 = create_study(direction=direction)
-    study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
+    save_static_image(figure)
+
+
+@parametrized_plot_contour
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+@pytest.mark.parametrize("num_studies", [0, 1, 2])
+def test_edf_plot_no_trials_studies(
+    plot_edf: Callable[..., Any], direction: str, num_studies: int
+) -> None:
+    studies = [create_study(direction=direction) for _ in range(num_studies)]
+    figure = plot_edf(studies)
+    save_static_image(figure)
+
+
+@parametrized_plot_contour
+@pytest.mark.parametrize("direction", ["minimize", "maximize"])
+@pytest.mark.parametrize("num_studies", [0, 1, 2])
+def test_edf_plot_target_label_studies(
+    plot_edf: Callable[..., Any], direction: str, num_studies: int
+) -> None:
+    studies = []
+    for _ in range(num_studies):
+        study = create_study(direction=direction)
+        study.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
+        studies.append(study)
+    figure = plot_edf(studies)
+
+    expected = "Objective Value"
+    if isinstance(figure, go.Figure):
+        assert figure.layout.xaxis.title.text == expected
+    elif isinstance(figure, Axes):
+        assert figure.xaxis.label.get_text() == expected
+
     with pytest.warns(UserWarning):
-        figure = plot_edf(study0, target=lambda t: t.params["x"])
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
+        figure = plot_edf(studies, target=lambda t: t.params["x"])
+    if isinstance(figure, go.Figure):
+        assert figure.layout.xaxis.title.text == expected
+    elif isinstance(figure, Axes):
+        assert figure.xaxis.label.get_text() == expected
 
     # Test with a customized target name.
-    study0 = create_study(direction=direction)
-    study0.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
-    figure = plot_edf(study0, target_name="Target Name")
-    _validate_edf_values(figure.data[0]["y"])
-    assert len(figure.data) == 1
-    assert figure.layout.xaxis.title.text == "Target Name"
+    figure = plot_edf(studies, target_name="Target Name")
+    expected = "Target Name"
+    if isinstance(figure, go.Figure):
+        assert figure.layout.xaxis.title.text == expected
+    elif isinstance(figure, Axes):
+        assert figure.xaxis.label.get_text() == expected
```

Comment:
I think we can split this test into different functions and reduces the complexity as follows.

```python
@parametrized_plot_edf
@pytest.mark.parametrize("direction", ["minimize", "maximize"])
@pytest.mark.parametrize("num_studies", [0, 1, 2])
def test_plot_edf_with_multiple_studies(
    plot_edf: Callable[..., Any], direction: str, num_studies: int
) -> None:
    studies = []
    for _ in range(num_studies):
        study = create_study(direction=direction)
        study.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
        studies.append(study)
    figure = plot_edf(studies)
    save_static_image(figure)


@parametrized_plot_edf
def test_plot_edf_with_target(plot_edf: Callable[..., Any]) -> None:
    study = create_study()
    study.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
    with pytest.warns(UserWarning):
        figure = plot_edf(study, target=lambda t: t.params["x"])
    save_static_image(figure)


@parametrized_plot_edf
@pytest.mark.parametrize("target_name", [None, "Target Name"])
def test_plot_edf_with_target_name(plot_edf: Callable[..., Any], target_name: Optional[str]) -> None:
    study = create_study()
    study.optimize(lambda t: t.suggest_float("x", 0, 5), n_trials=10)
    figure = plot_edf(study, target_name=target_name)

    expected = target_name if target_name is not None else "Objective Value"    
    if isinstance(figure, go.Figure):
        assert figure.layout.xaxis.title.text == expected
    elif isinstance(figure, Axes):
        assert figure.xaxis.label.get_text() == expected

    save_static_image(figure)
```

Reply:
@contramundum53  Sorry, the link looks broken. Could you give a link again?